packages/geo/src/lib/print/shared/print.service.ts

```diff
@@ -1335,4 +1337,176 @@ export class PrintService {
+    // convert icon list to base64
+    let iconsArray: Array<{name: string, icon: string}> = [];
+    for (let index = 0; index < matListItem.length; index++) {
+      const element = matListItem[index];
+      const icon = element.getElementsByTagName('mat-icon')[0] as HTMLElement;
+      const iconName = icon.getAttribute('data-mat-icon-name') as string;
+      const found = iconsArray.some(el => el.name === iconName);
+      if(!found) {
+        const iconCanvas = await html2canvas(icon, { scale: 3});
+        iconsArray.push({name: iconName, icon: iconCanvas.toDataURL()});
+      }
+    }
+
+    let formattedDirection: Array<{ instruction: string, icon: string, distance: string }> = [];
+    for (let i = 0; i < direction.steps.length; i++) {
+      const step = direction.steps[i];
+      const instruction = formatInstruction(
+        step.maneuver.type,
+        step.maneuver.modifier,
+        step.name,
+        step.maneuver.bearing_after,
+        i,
+        (step.maneuver as any).exit,
+        this.languageService,
+        i === direction.steps.length - 1
+      );
+
+      const distance = formatDistance(step.distance);
+      formattedDirection.push({
+        instruction: (i+1)+'. '+instruction.instruction,
+        icon: iconsArray.find((icon) => icon.name === instruction.image).icon,
+        distance:  (distance) ? '('+ distance +')' : undefined
+      });
+    }
+    return formattedDirection;
+  }
+
+  downloadDirection(
+    map: IgoMap,
+    direction: Direction
+  ): Observable<SubjectStatus> {
+
+    const status$ = new Subject<SubjectStatus>();
+
+    const doc: any = new jsPDF({
+      orientation: 'p',
+      format: 'Letter',
+      unit: 'mm' // default
+    });
+
+    const size = map.ol.getSize();
+    map.ol.once('rendercomplete', async (event: any) => {
+      const mapCanvas = event.target.getViewport().getElementsByTagName('canvas') as HTMLCollectionOf<HTMLCanvasElement>;
+      const mapResultCanvas = await this.drawMap(size, mapCanvas);
+      this.renderMap(map, size, 5);
+      await this.drawMapControls(map, mapResultCanvas, PrintLegendPosition.none);
+
+      const imgSize = this.getImageSizeToFitPdf(doc, mapResultCanvas, [10, 10, 10, 10]);
+
+      doc.addImage(mapResultCanvas, 10, 20, imgSize[0], imgSize[1]);
+      doc.rect(10, 20, imgSize[0], imgSize[1]);
+
+      doc.setFontSize(14);
+      const title = `${direction.title} (${formatDistance(direction.distance)}, ${formatDuration(direction.duration)})`;
+      const titleP = (imgSize[1] + 10 + 20);
+      doc.text(title, (doc.internal.pageSize.width/2) ,titleP, {align: 'center'});
+
+      const HTMLtable = await this.setHTMLTableContent(direction);
+      const tablePos = titleP + 5;
+      const totalPagesExp = '{total_pages_count_string}';
+      const date = moment(Date.now()).format("DD/MM/YYYY hh:mm").toString();
+      const appName = (this.configService.getConfig('title')) ?
+      this.configService.getConfig('title') : 'IGO Lib';
```

Comment:
S'il n'y a pas de titre, je crois qu'on pourrait ne rien mettre. 
On avait parlé de mettre un logo a cet endroit? 

On a [une issue](https://github.com/infra-geo-ouverte/igo2-lib/issues/1328) qui discute d'une stratégie pour ajouter un logo. Si on veut mettre un logo, on pourrait ajouter une tâche dans l'issue #1328 pour venir complété le travaille ici.

Reply:
TO DO: customization for page header logo

Reply:
FInalement le ticket #1328 ne couvrait pas vraiment ton besoin. Il faudrait voir pour vérifier la présence ou ajouter une config pour le logo et l'ajotuer ici.